### ISSUE TITLE
Node deleted in the backend is seen stuck in registering state in Rancher

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#32302]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
- Deploy a node driver DO cluster. - Two nodepools with same name
```
sowmya-cis-1 --> etcd/control plane - 1 node
sowmya-cis-1 --> worker role - 1 node
```
- worker node is seen stuck in registering state.
- Scale up the worker node - `sowmya-cis-3` is added.
<img width="1014" alt="Screen Shot 2021-04-22 at 4 05 06 PM" src="https://user-images.githubusercontent.com/26032343/115796842-65470e00-a387-11eb-8eb4-054ce3c30fc3.png">

**Other details that may be helpful:**

- This is not happening always.
- Reach out t me for rancher server logs

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.8-rc7
- Installation option (single install/HA): HA AKS local cluster

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): RKE DO node driver
- Kubernetes version (use `kubectl version`):

```
1.20 (Any)
```

**Comment by sowmyav27** [comment]
**See this issue on master-head commit id: `5d49b92`**

- Deploy a DO RKE1 cluster from Ember UI with 2 nodepools
```
sowmya-issue-1 --> etcd/control plane - 1 node
sowmya-issue-1 --> worker role - 1 node
```

- Worker node gets the IP and master node fails with error `Error creating machine: Error detecting OS: Too many retries waiting for SSH to be available. Last error: Maximum number of retries (60) exceeded; lstat /opt/jail/c-hv456/management-state/node/nodes/sowmya-issue-1: no such file or directory` and re provisions.
- When it is re-provisioning the worker node is deleted in the backend instead of the control plane node. (both have the same nodename)
- Worker node is seen stuck in Registering state in Rancher UI

**Expected Result:**
Worker node should not be deleted and the master node should be from the backend.

**Note:**
Happens always with ^^^ these steps

**Comment by deniseschannon** [comment]
We could add in validation to prevent having node pools with the same name to prevent getting into this scenario.

**Comment by snasovich** [comment]
@deniseschannon , are we OK to implement the approach you've suggested or it needs approval by PM or something of sorts?

**Comment by deniseschannon** [comment]
No approval needed from PM as node pools are a Rancher resource so there is no true need for duplicate names.

**Comment by cmurphy** [comment]
I think this validation will have to happen in UI. The cluster create request does not send the nodepool names in a single cluster request, it sends separate requests for the cluster and each nodepool:

![image](https://user-images.githubusercontent.com/1220710/125000819-11eb7000-e006-11eb-869b-691c981dfe2d.png)

so the cluster validator in Rancher does not know the node pool names, and the node pool validator does not know the name of the other node pool request. However, the UI already does some nodepool validation:

https://github.com/rancher/ui/blob/ea36d73a762f8c4e93d21f1019c6de8138df5d3c/lib/shared/addon/components/cru-node-pools/component.js#L188-L227

It has all the node pool names on the form so it has all the information needed to make sure the node pool names are unique for a cluster create request.

It's possible when running in CATTLE_DEV_MODE to get into a similar state with two separate clusters using the same node pool name. But in non-dev mode, the state directory includes the cluster ID so it's impossible for the state of two node pools in different clusters to overlap:

https://github.com/rancher/rancher/blob/08b9f8baf424664e1637bd42c1a1f3a3aa9ade3d/pkg/nodeconfig/config.go#L308 

If we wanted to address this it could be done in backend, but IMO it's not important to add this cross-cluster naming restriction only to make dev mode safe.

**Comment by snasovich** [comment]
@nwmac @gaktive , please take a look at the above comment. Can this be added to UI team backlog to get addressed?

**Comment by gaktive** [comment]
@snasovich we can take a look. Is there a targeted release for this since this is presently unscheduled? Sounds like a 2.5.x but if this is needed for 2.6.0, then we'd need to jump on this quick.

**Comment by snasovich** [comment]
@gaktive , no, this is a rather low priority at this time (going to tentatively assign v2.6.2 milestone). I just wanted to get it handed off to UI team so it doesn't sit assigned to my team.
